core/server.go

```diff
@@ -72,3 +94,68 @@ func Server(ctx context.Context, name string, hl Manager, grpcHl GrpcManager, lo
 
 	return <-errChan
 }
+
+type ServerOption interface {
+	Apply(params *ServerParams) error
```

Comment:
Change function signature to `apply(params *ServerParams)` so that:
- we're not exporting the function, and 
- error checking can be done in `Start`